crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -502,6 +527,102 @@ impl<'a> TimelineEventHandler<'a> {
         self.state.reactions.map.insert(reaction_id, (reaction_sender_data, c.relates_to));
     }
 
+    #[instrument(skip_all, fields(replacement_event_id = ?replacement.event_id))]
+    fn handle_poll_start_edit(
+        &mut self,
+        replacement: Replacement<UnstablePollStartEventContentWithoutRelation>,
+    ) {
+        update_timeline_item!(self, &replacement.event_id, "poll edit", |event_item| {
+            if self.ctx.sender != event_item.sender() {
+                info!(
+                    original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                    "Edit event applies to another user's timeline item, discarding"
+                );
+                return None;
+            }
+
+            let TimelineItemContent::Poll(poll_state) = &event_item.content() else {
+                        info!(
+                            original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                            "Can't edit a poll that is not of type TimelineItemContent::Poll, discarding"
+                        );
+                        return None;
+                    };
```

Comment:
Indented too much.

Reply:
Fixed. Not sure why most of the times vscode applies formatting automatically but sometimes not.

Reply:
In this case that's probably because you don't have a recent enough version of rustfmt on your system. Iirc, support for formatting let-else statements has been added in one of the very latest versions.